src/main/java/org/broadinstitute/hellbender/tools/picard/vcf/GatherVcfs.java

```diff
@@ -91,8 +91,7 @@ protected Object doWork() {
     /** Checks (via filename checking) that all files appear to be block compressed files. */
     private boolean areAllBlockCompressed(final List<File> input) {
         for (final File f : input) {
-            // TODO move/expose these methods elsewhere, since VariantContextWriterFactory is deprecated
-            if (VariantContextWriterFactory.isBCFOutput(f) || !VariantContextWriterFactory.isCompressedVcf(f)) return false;
+            if ((f != null && f.getName().contains(".bcf")) || !AbstractFeatureReader.hasBlockCompressedExtension(f)) return false;
```

Comment:
I think that we lost a null check that was being done by `VariantContextWiterFactory.isCompressedVcf()`.  This could possibly introduce new `NullPointerException`s (NPE) where there weren't any before.